libbeat/publisher/queue/queue.go

```diff
@@ -122,3 +127,15 @@ type Batch interface {
 	Events() []publisher.Event
 	ACK()
 }
+
+// AdjustInternalQueueSize decides the size for the internal queue used by most queue implementations.
+func AdjustInternalQueueSize(requested, mainQueueSize int) (actual int) {
```

Comment:
Actually, sorry for the shuffle but I think I'd prefer if this was back within `memqueue`... this calculation is nearly a noop for the spool, which behaves very differently (and will be going away in the future anyway), so the real use of this helper is just to cap the intake size for the memory queue to something reasonable.

In an ideal world I'd actually like the flag itself to be in `memqueue.Settings` (like the analogous `WriteAheadLimit` flag is on the disk queue), but right now there's no clean way to do that without exposing the setting to the user, which seems undesirable. I don't want to gate the change on fixing that plumbing, so it makes sense to me to put it on the Beat config, but I'd still prefer to keep the scope narrow if possible.

Reply:
I moved this to memque. Also renamed all references to "InternalQueueSize" to "InputQueueSize", which I think it better defines it, "internal queue" being what we call the main event queue (mem, spool...) in our docs.